lib/stat_tracker.rb

```diff
@@ -34,6 +34,38 @@ def lowest_total_score
   end
 
 #------------LeagueStatistics
+
+
+#------------SeasonStatistics
+
+  def coach_list_wins_losses
+    @coach_hash = Hash.new
+    game_teams.each do |gt|
+      (@coach_hash[gt.head_coach] ||= []) << gt.result
+    end
+    @coach_hash
+  end
+
+  def winningest_coach
+    winner = @coach_hash.max_by do |key, w_l|
+      wins = w_l.count("WIN")
+      losses = w_l.count("LOSS").to_f
+      (wins) / (wins + losses)
+    end
+    winner[0]
+  end
+
+  def worst_coach
+    loser = @coach_hash.min_by do |key, w_l|
+      wins = w_l.count("WIN")
+      losses = w_l.count("LOSS").to_f
+      (wins) / (wins + losses)
+    end
+    loser[0]
+  end
```

Comment:
Once the instance variables are gone you can just call coach_list_wins_losses and then iterate over that!
 
So it'd look like this:
```
winner = coach_list_wins_losses.max_by do |key, w_l|
loser = coach_list_wins_losses.min_by do |key, w_l|
```